app/views/docs/authentication-oauth.phtml

```diff
@@ -0,0 +1,370 @@
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="appwrite-callback-[PROJECT_ID]" />
+      </intent-filter>
+    </activity>
+  </application>
+</manifest>'); ?>
+</code></pre>
+</div>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import io.appwrite.Client
+import io.appwrite.services.Account
+
+val client = Client()
+    .setEndpoint("https://cloud.appwrite.io/v1") // Your API Endpoint
+    .setProject("[PROJECT_ID]")                // Your project ID
+
+val account = Account(client)
+
+// Go to OAuth provider login page
+account.createOAuth2Session(provider = "amazon")</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Apple</h3>
+        <p>In order to capture the Appwrite OAuth callback url, the following URL scheme needs to added to your <code>Info.plist</code>.</p>
+
+<div class="ide" data-lang="html" data-lang-label="XML">
+    <pre class="line-numbers"><code class="prism language-xml" data-prism><?php echo $this->escape('<key>CFBundleURLTypes</key>
+<array>
+<dict>
+    <key>CFBundleTypeRole</key>
+    <string>Editor</string>
+    <key>CFBundleURLName</key>
+    <string>io.appwrite</string>
+    <key>CFBundleURLSchemes</key>
+    <array>
+        <string>appwrite-callback-[PROJECT_ID]</string>
+    </array>
+</dict>
+</array>
+');?></code></pre>
+</div>
+
+<p>If you're using UIKit, you'll also need to add a hook to your <code>SceneDelegate.swift</code> file to ensure cookies work correctly.</p>
+
+<div class="ide" data-lang="swift" data-lang-label="Swift">
+  <pre class="line-numbers"><code class="prism language-swift" data-prism><?php echo $this->escape('
+func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+    guard let url = URLContexts.first?.url,
+        url.absoluteString.contains("appwrite-callback") else {
+        return
+    }
+    WebAuthComponent.handleIncomingCookie(from: url)
+}
+');?></code></pre>
+</div>
+        <div class="ide" data-lang="swift" data-lang-label="Apple SDK">
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+
+let client = Client()
+    .setEndpoint("https://cloud.appwrite.io/v1") // Your API Endpoint
+    .setProject("[PROJECT_ID]")                // Your project ID
+
+let account = Account(client)
+
+// Go to OAuth provider login page
+try await account.createOAuth2Session(provider: "amazon")</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>GraphQL</h3>
+        <div class="notice">
+            <p>OAuth 2 is not available through the GraphQL API. You can use the REST API or any Client SDK instead.</p>
+        </div>
+    </li>
+</ul>
+
+<p>
+    You'll be redirected to the OAuth 2 provider's login page to log in. 
+    Once complete, you'll be redirected to the redirect URL configured in your OAuth 2 provider.
```

Comment:
This is internal complexity (it redirect to appwrite API)
We could say:

Once complete, you will be redirected back to Appwrite API where user and session is created. User is next (almost instantly) redirected back to your application.

Reply:
Done